### ISSUE TITLE
MonadIO instance is inconveniently specific

### ISSUE THREAD

**Issue by robrix** [robrix/fused-effects#98]
It’s inconvenient that the `MonadIO` instance for `Eff` requires `Lift IO` and not `MonadIO m => Lift m`. Unfortunately, we can’t generalize it to the latter because the instance head doesn’t mention `m` and thus it would be ambiguous. (`AllowAmbiguousTypes` would resolve that, but the ergonomics would be dreadful—any call to `liftIO` in an `Eff` context would have to deal with the ambiguity, requiring a lot of extra type annotations.)

We could (potentially) resolve this with some sort of `LastMember` constraint on the signature, since it could be uniquely determined by the signature and the signature is uniquely determined by the `Carrier` constraint.

Alternatively, if the continuation of our recent optimization/benchmarking work (cf #90) results in the simplification, deprioritization, or even removal of the `Eff` type, then we’d likely define `MonadIO` directly on the carriers, which would have the same result.

cf #82
cf https://github.com/robrix/fused-effects/issues/96#issuecomment-453838137

**Comment by robrix** [comment]
Of the two, I prefer `LastMember`, as it maintains the clean separation between interface (effect) and implementation (carrier), and avoids proliferation of _n²_ instances. IIRC `freer-simple` does something like this.

**Comment by deepfire** [comment]
@robrix, does the following signature for `sendM` seem appropriate:
```
sendM :: (Monad m, Monad (effect m), Member (Lift (effect m)) sig, Member effect sig, Carrier sig m) => effect m a -> m a
sendM m = send (Lift (fmap pure m))
```
?

**Comment by robrix** [comment]
This was fixed in #111; this just goes away when we piggyback on typeclass resolution.
